plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/storage/KVMStorageProcessor.java

```diff
@@ -1166,6 +1167,14 @@ protected synchronized String attachOrDetachDevice(final Connect conn, final boo
             } else {
                 s_logger.debug("Detaching device: " + xml);
                 dm.detachDevice(xml);
+                LibvirtDomainXMLParser parser = new LibvirtDomainXMLParser();
+                parser.parseDomainXML(dm.getXMLDesc(0));
+                List<DiskDef> disks = parser.getDisks();
+                for (DiskDef diskDef : disks) {
+                    if (StringUtils.contains(xml, diskDef.getDiskPath())) {
```

Comment:
@slavkap can you check if this is applicable for attach disk as well (when VM is either booting or shutting down)?

Reply:
Sure, @sureshanaparti, I'll do the test and will let you know.

Reply:
@sureshanaparti, sorry for the late response, but I didn't have time to check it yesterday! The change is applied only to detach a volume when the VM is in Running or Stopped state